pkg/scheduler/scheduler_test.go

```diff
@@ -1723,3 +1730,167 @@ func TestRequeueAndUpdate(t *testing.T) {
 		})
 	}
 }
+
+func TestHeads(t *testing.T) {
```

Comment:
can we keep this where it was and just use a fake cache? The unit test should be in the pkg where the function lives.